exporter/awsxrayexporter/translator/http.go

```diff
@@ -0,0 +1,290 @@
+// Copyright 2019, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package translator
+
+import (
+	"net/http"
+	"strconv"
+
+	tracepb "github.com/census-instrumentation/opencensus-proto/gen-go/trace/v1"
+	tracetranslator "github.com/open-telemetry/opentelemetry-collector/translator/trace"
+)
+
+// OpenTelemetry Semantic Convention attribute names for HTTP and gRPC related attributes
+const (
+	MethodAttribute                  = "http.method"
+	URLAttribute                     = "http.url"
+	TargetAttribute                  = "http.target"
+	HostAttribute                    = "http.host"
+	SchemeAttribute                  = "http.scheme"
+	StatusCodeAttribute              = "http.status_code"
+	StatusTextAttribute              = "http.status_text"
+	FlavorAttribute                  = "http.flavor"
+	ServerNameAttribute              = "http.server_name"
+	PortAttribute                    = "http.port"
+	RouteAttribute                   = "http.route"
+	ClientIPAttribute                = "http.client_ip"
+	UserAgentAttribute               = "http.user_agent"
+	MessageTypeAttribute             = "message.type"
+	MessageIDAttribute               = "message.id"
+	MessageCompressedSizeAttribute   = "message.compressed_size"
+	MessageUncompressedSizeAttribute = "message.uncompressed_size"
+)
+
+// HTTPData provides the shape for unmarshalling request and response data.
+type HTTPData struct {
+	Request  RequestData  `json:"request,omitempty"`
+	Response ResponseData `json:"response,omitempty"`
+}
+
+// RequestData provides the shape for unmarshalling request data.
+type RequestData struct {
+	Method        string `json:"method,omitempty"`
+	URL           string `json:"url,omitempty"` // http(s)://host/path
+	ClientIP      string `json:"client_ip,omitempty"`
+	UserAgent     string `json:"user_agent,omitempty"`
+	XForwardedFor bool   `json:"x_forwarded_for,omitempty"`
+	Traced        bool   `json:"traced,omitempty"`
+}
+
+// ResponseData provides the shape for unmarshalling response data.
+type ResponseData struct {
+	Status        int64 `json:"status,omitempty"`
+	ContentLength int64 `json:"content_length,omitempty"`
+}
+
+func convertToHTTPStatusCode(code int32) int64 {
```

Comment:
There is already a mapping from HTTP to gRPC in `github.com/open-telemetry/opentelemetry-collector/translator/trace` package. It would be best to add the corresponding reverse translation that is done by this function to that same package for consistency.
Will you be able to create a separate PR for that at https://github.com/open-telemetry/opentelemetry-collector/ ?

Reply:
Moved to collector